numpy_challenge_py (1).py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""numpy_challenge.py
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1XHEJneVVUSc4H8NSkeE70HMY8rG5f5Rc
+
+### 1 Cоздать 3 ваших любимых эррея разными способами
+"""
+
+import numpy as np
+
+#1 преобразовать список в массив нампай
+food = ['banana', 'coffee', 'tiramisu', 'pasta', 'pizza']
+food = np.array(food)
+
+food
+
+#2 cделать массив через np.arange
+
+birthdays_days = np.arange(2,100, 12)
+
+birthdays_days
+
+#3 Создать с нуля массив нампай
+family = np.array(['Masha', 'Varya', 'Papa', 'Mama', 'Babuska', 'Dedushka', 'Dyada Yura'])
+
+family
+
+"""## 2. Функция для проверки возможности умножения матриц"""
+
+def multiplication_check(matrix_list):
+  matrix1 = matrix_list[0]
+  matrix2 = matrix_list[1]
+  #умножение двух неодномерных или первого неодномерного и второго одномерного
+  if (np.array(matrix1).ndim != 1 and np.array(matrix2).ndim != 1) or (np.array(matrix1).ndim != 1 and np.array(matrix2).ndim == 1):
+    if np.shape(matrix1)[1] ==  np.shape(matrix2)[0]:
+      return True
+    else:
+      return False
+  #когда первый одномерный, а второй неодномерный
+  elif np.array(matrix1).ndim == 1 and np.array(matrix2).ndim != 1:
+    if np.array(matrix1).shape[0] == np.array(matrix2).shape[0]:
+      return True
+    else:
+      return False
+  elif np.array(matrix1).ndim == 1 and np.array(matrix2).ndim == 1:
+    if np.array(matrix1).size == 1 and np.array(matrix2).size == 1:
+      return True
+    else:
+      return False
+
+"""## 3. Функция для умножения"""
+
+def matrix_multiplication(matrix1,matrix2):
+  if multiplication_check([matrix1, matrix2]) == True:
+    product = np.array(matrix1).dot(np.array(matrix2))
+    return product
+  else:
+    print('The matrices cannot be multiplied')
```

Comment:
1. Отступ - 4 пробела
2. Аналогично предыдущему заданию, не нужно везде писать `np.array`, если это можно сделать один раз

```suggestion
def matrix_multiplication(matrix1, matrix2):
    if multiplication_check([matrix1, matrix2]):
        product = np.array(matrix1).dot(np.array(matrix2))
        return product
  else:
    print('The matrices cannot be multiplied')
```